packages/opentelemetry-metrics/src/Meter.ts

```diff
@@ -231,20 +230,14 @@ export class Meter implements api.Meter {
    * each aggregator belonging to the metrics that were created with this
    * meter instance.
    */
-  collect(): Promise<unknown> {
-    return new Promise((resolve, reject) => {
-      const metrics: Promise<MetricRecord[]>[] = [];
-      Array.from(this._metrics.values()).forEach(metric => {
-        metrics.push(metric.getMetricRecord());
+  collect(): Promise<void> {
+    const metrics = Array.from(this._metrics.values()).map(metric => {
+      return metric.getMetricRecord();
+    });
+    return Promise.all(metrics).then(records => {
```

Comment:
Return type is `Promise<void>`, but this returns `Promise<Array<void>>`. A minor distinction, but we should make the return match the types. Making this function `async` and using `await` here should do the trick.

Going to mark as ✅ since it is a minor nit.